sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -133,8 +136,27 @@ private object CosmosConfig {
     userProvidedOptions: Map[String, String] // user provided config
   ) : Map[String, String] = {
 
+    var accountDataResolverCls: AccountDataResolver = null
+    val serviceLoader = ServiceLoader.load(classOf[AccountDataResolver])
+    val iterator = serviceLoader.iterator()
+    if (iterator.hasNext()) {
+        accountDataResolverCls = iterator.next()
+    }
+
+    var accountDataConfig : Map[String, String] = null
+    if (accountDataResolverCls != null) {
```

Comment:
we need tests for this feature.

Please add a unit test for this, see this file for example of similar unit tests:
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/CosmosConfigSpec.scala

Reply:
As discussed offline, will be fixing the test cases outside this PR.